core/src/main/java/hudson/Util.java

```diff
@@ -1662,7 +1665,27 @@ public static int permissionsToMode(Set<PosixFilePermission> permissions) {
             throw new IOException(e);
         }
     }
-
+    
+    /**
+     * Compute the number of calendar days elapsed since the given date.
+     * As it's only the calendar days difference that matter, "11.00pm" to "2.00am the day after" returns 1,
+     * even if there are only 3 hours between. As well as "10am" to "2pm" both on the same day, returns 0.
+     */
+    @Restricted(NoExternalUse.class)
+    public static int differenceInCalendarDay(@Nonnull Date a, @Nonnull Date b){
+        LocalDate aLocal = a.toInstant().atZone(ZoneId.systemDefault()).toLocalDate();
+        LocalDate bLocal = b.toInstant().atZone(ZoneId.systemDefault()).toLocalDate();
+        return Period.between(aLocal, bLocal).getDays();
```

Comment:
I think `Period#getDays` just returns the number of days less than a month, not the total (i.e. differenceInCalendarDay(2017-01-01, 2018-01-01) will return 0, but I think you want it to return 365). You should use `ChronoUnit.DAYS.between(aLocal, bLocal)` to get the total number of days.

Reply:
Nice catch! It's due to things like this that I tend to use `Instant` more often than the human versions.